src/commands.cpp

```diff
@@ -8223,40 +8258,46 @@ bool CommandBackupPut::procresult(Result r)
         e = r.errorOrOK();
     }
 
-    if (!mUpdate && !ISUNDEF(backupId)) // successfully registered a new backup --> set the backup name
+    // Upon new backup successfully registered --> set the backup name silently
+    if (!mUpdate && !ISUNDEF(backupId))
     {
         assert(r.succeeded());
 
-        // automatically add the backup name to the corresponding user attribute
         std::string key {Base64Str<MegaClient::BACKUPHANDLE>(backupId)};
-        attr_t attrtype = ATTR_BACKUP_NAMES;
-
-        std::unique_ptr<TLVstore> tlv;
+        attr_t attrType = ATTR_BACKUP_NAMES;
 
         User *ownUser = client->finduser(client->me);
-        const std::string *oldValue = ownUser->getattr(attrtype);
-        if (!oldValue)  // attr doesn't exist -> create it
-        {
-            tlv.reset(new TLVstore());
-            tlv->set(key, mBackupName);
+        const std::string *oldValue = ownUser->getattr(attrType);
 
-            // serialize and encrypt the TLV container
-            std::unique_ptr<std::string> container(tlv->tlvRecordsToContainer(client->rng, &client->key));
-            client->putua(attrtype, (byte *)container->data(), unsigned(container->size()));
+        if (oldValue && !ownUser->isattrvalid(attrType)) // not fetched yet or outdated
+        {
+            LOG_warn << "Cannot immediately set backup name for backup id: " << backupId << ". Fetching...";
+            client->getua(ownUser, attrType, 0);
+            client->mSendingBackupName = true;
         }
-        else if (!ownUser->isattrvalid(attrtype)) // not fetched yet or outdated
+
+        if (client->mSendingBackupName)
         {
-            LOG_err << "Failed to set backup name for backup id : " << backupId;
+            // accumulate this update for the future, in order to avoid race conditions
+            // they will be sent upon `upv` completion for the update in progress
+            client->mPendingBackupNames[key] = mBackupName;
         }
         else
         {
-            string_map attrMap { {key, Base64::btoa(mBackupName)} };
-            tlv.reset(TLVstore::containerToTLVrecords(oldValue, &client->key));
-            if (User::mergeUserAttribute(attrtype, attrMap, *tlv.get()))
+            // send backup name for this backup directly
+            std::unique_ptr<TLVstore> tlv { !oldValue
+                        ? new TLVstore()
+                        : TLVstore::containerToTLVrecords(oldValue, &client->key) };
+
+            string_map attrMap = move(client->mPendingBackupNames);
```

Comment:
mPendingBackupNames would be empty here, wouldn't it?

Reply:
Not necessarily. First time, yes, but later it could happen that we just finished sending pending backup names and we got some new pending backup names.